From 67c11ca4793966e8246122af554ec05999a158af Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Mon, 14 Dec 2009 16:19:29 -0500
Subject: [PATCH 2/8] io_apic: delete duplicate definitions, use header file

The definitions of these io_apic structs existed in two
places; the source file, and an unused header file.  The
definitions are the same, so delete the source file ones,
and use the header file ones.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
---
 kernel/vbi/io_apic.c |   76 +-------------------------------------------------
 1 files changed, 1 insertions(+), 75 deletions(-)

diff --git a/kernel/vbi/io_apic.c b/kernel/vbi/io_apic.c
index 469b1bc..29ded32 100644
--- a/kernel/vbi/io_apic.c
+++ b/kernel/vbi/io_apic.c
@@ -17,6 +17,7 @@
 #include <linux/types.h>
 #include <linux/module.h>
 #include <vbi/vbi.h>
+#include <vbi/io_apic.h>
 #include <vbi/private.h>
 
 /*
@@ -53,81 +54,6 @@
 	(*((volatile uint32_t *)(((unsigned long *)(base)) + (offset))) = \
 	(value))
 
-/* data structures */
-
-/* IO APIC ID register */
-typedef union
-{
-	struct
-	{
-		uint32_t reserved1:24;	/* reserved bits */
-		uint32_t IOAPIC_id:4;	/* apic id */
-		uint32_t reserved2:4;	/* reserved bits */
-	} field;
-	uint32_t value;
-} VIOAPIC_ID;
-
-/* IO APIC Version Register */
-typedef union
-{
-	struct
-	{
-		uint32_t version:8;	/* version identifier */
-		uint32_t reserved1:8;	/* reserved bits */
-		uint32_t maxRedirEntry:8;/* maximum number of entries - 1, or
-					 * number of IRQ pins - 1 */
-		uint32_t reserved2:8;	/* reserved bits */
-	} field;
-	uint32_t value;
-} VIOAPIC_VERSION;
-
-/* Virtual IO APIC redirection table entry. Split into high/low 32 */
-typedef union
-{
-	struct
-	{
-		uint32_t reserved:24;	/* reserved bits */
-		uint32_t destination:8;	/* destination field */
-	} field;
-	uint32_t value;
-} VIOAPIC_REDIR_HIGH;
-
-typedef union
-{
-	struct
-	{
-		uint32_t vector:8;	/* vector number */
-		uint32_t deliveryMode:3;/* delivery mode:
-					 * fixed, lowest, SMI, reserved, NMI,
-					 * INIT, reserved, extInt */
-		uint32_t destMode:1;	/* dest. mode: 0=physical, 1=logical */
-		uint32_t delivStatus:1;	/* delivery status: 0=idle, 1=pending*/
-		uint32_t polarity:1;	/* polarity: 0=high active */
-		uint32_t irr:1;		/* remote IRR (level only): 1 before
-					 * LAPIC accepts level, 0 after EOI */
-		uint32_t trigger:1;	/* 0=edge, 1=level */
-		uint32_t mask:1;	/* 1=masked */
-		uint32_t reserved:15;	/* reserved bits */
-	} field;
-	uint32_t value;
-} VIOAPIC_REDIR_LOW;
-
-typedef struct vioapicEntry
-{
-	VIOAPIC_REDIR_LOW vioapicLow;
-	VIOAPIC_REDIR_HIGH vioapicHigh;
-} VIOAPIC_ENTRY;
-
-typedef struct vioapic
-{
-	VIOAPIC_ID id;
-	VIOAPIC_VERSION version;
-	uint32_t arb;
-	uint32_t boot;
-	uint32_t reserved[6];
-	VIOAPIC_ENTRY entry [128];
-} VIOAPIC;
-
 /*
  * vbi_get_vioapic_addr - Get VIOAPIC base address
  *
-- 
1.6.5.2

